presto-main/src/main/java/com/facebook/presto/sql/gen/RowExpressionCompiler.java

```diff
@@ -87,23 +87,20 @@ public BytecodeNode compile(RowExpression rowExpression, Scope scope, Optional<V
         @Override
         public BytecodeNode visitCall(CallExpression call, Context context)
         {
-            BytecodeGenerator generator;
-            // special-cased in function manager
-            if (call.getSignature().getName().equals(CAST)) {
```

Comment:
yes, we should refactor `CAST` as well...

Reply:
Check out the next diff ;)